reframe/frontend/cli.py

```diff
@@ -475,6 +477,8 @@ def main():
             mode_args = site_config.get(f'modes/@{options.mode}/options')
 
             # Parse the mode's options and reparse the command-line
+            mode_args = list(itertools.chain(*[shlex.split(m)
```

Comment:
We will need a small comment here as well as of why we have to lexically split each option.